test/unit/types/json_type_test.rb

```diff
@@ -1,4 +1,20 @@
 require 'test_helper'
 
 class Superstore::Types::JsonTypeTest < Superstore::Types::TestCase
+  test 'serializes empty array to nil' do
+    comments = { 'foo' => 'bar' }
+    issue1   = Issue.new(comments:)
+    issue2   = Issue.new(comments: [])
+    issue3   = Issue.new(comments: {})
+
+    assert_equal comments, issue1.comments
+    assert_equal [], issue2.comments
+    assert_equal({}, issue3.comments)
+
+    [issue1, issue2].each(&:save!).each(&:reload)
+
+    assert_equal comments, issue1.comments
+    assert_nil issue2.comments
+    assert_equal({}, issue3.comments)
```

Comment:
I'm not sure why `{}` is not being serialized to `nil` as well, but unless we have a specific reason why we want it to be otherwise, I think this behavior is fine here.

Reply:
Is it because issue 3 isn't saved above? Isn't the goal to set empty objects to nil?

Reply:
Yes, thank you. 7c7eb50